tsdb/engine/tsm1/encoding.go

```diff
@@ -307,11 +307,11 @@ func DecodeFloatBlock(block []byte, a []*FloatValue) ([]*FloatValue, error) {
 	for dec.Next() && iter.Next() {
 		ts := dec.Read()
 		v := iter.Values()
-		if i < len(a) && a[i] != nil {
+		if i < len(a) {
 			a[i].time = ts
 			a[i].value = v
 		} else {
-			a = append(a, &FloatValue{ts, v})
+			a = append(a, FloatValue{ts, v})
```

Comment:
Should probably update the other types as well.  I would expect them to have similar issues, right?